### ISSUE TITLE
Trouble inside callback

### ISSUE THREAD

**Issue by dreyliky** [if1live/socket.io-unity-sample#2]
Hello. I have some strange trouble and I so confused on this.
I can't do a lot of actions inside WS event callback.

For example, one specific thing that I was trying to do, is SceneManager.LoadScene(SOME_SCENE) inside callback. Callback method "breaks off" on this line... I don't have any exceptions or errors. It just "ignore" this method execution and "breake off". It's so strange for me :c

I will really grateful if you know reason of this behavior.

this._Socket.On(Socket.EVENT_CONNECT, () => {
// WORKS
Debug.Log("CONNECTED");

// There is no any exception or erros! Method just breaks off  
SceneManager.LoadScene(SceneEnum.Game);

// INGORING
Debug.Log("Ok");
});

P.S. If I execute my SceneManager.LoadScene(SceneEnum.Game); for example in Awake or Start method it obviously will work great...

I also tryed emit event inside callback using UniRx Subject and subscribe on it in different place and execute LoadScene but there was the same behaviour and I'm out of my mind now!!!

**Comment by goyusia** [comment]
@dreyliky maybe `Can only be called from the main thread` problem.

If you use use UniRx, use this code
```cs
subject.AsObservable().ObserveOnMainThread().Subscribe(xs => {
  SceneManager.LoadScene(SceneEnum.Game);
})
```

**Comment by dreyliky** [comment]
@if1live Thanks a lot! You are my hero TWICE! :D
I'm frontend developer and I don't work with C# a lot, so I spent about 2 days to find some C# SocketIO Client library for Unity and connect it to my NodeJs server... I'm tryed about 7 variants and all of them didn't work as i needed... And then I found your repository and it started work great!!! But then I discovered this problem and you also helped me. So thank you a lot! C:
